src/modules/mavlink/mavlink_receiver.h

```diff
@@ -314,6 +315,26 @@ class MavlinkReceiver : public ModuleParams
 	// Allocated if needed.
 	TunePublisher *_tune_publisher{nullptr};
 
+	hrt_abstime _last_heartbeat_check{0};
+
+	hrt_abstime _heartbeat_type_antenna_tracker{0};
```

Comment:
You could also store them as bits and clear all of them every 1-1.5s. The max delay for loss detection would be even higher though.

Reply:
I think that would be fine as well. We can strip this down to bits later if it's going to keep growing (more supported MAV_TYPEs/MAV_COMPONENTs).